web/client/components/map/openlayers/MeasurementSupport.jsx

```diff
@@ -49,12 +51,49 @@ class MeasurementSupport extends React.Component {
     };
 
     componentWillReceiveProps(newProps) {
-        if (newProps.measurement.geomType && newProps.measurement.geomType !== this.props.measurement.geomType ) {
+        this.invalidCoordinates = [];
+        if (newProps.measurement.geomType && newProps.measurement.geomType !== this.props.measurement.geomType ||
+            /* check also when a default is set
+             * if so the first condition does not match
+             * because the old geomType is not changed (it was laready defined as default)
+             * and the measure tool is getting enabled
+            */
+            (newProps.measurement.geomType && this.props.measurement.geomType && (newProps.measurement.lineMeasureEnabled || newProps.measurement.areaMeasureEnabled || newProps.measurement.bearingMeasureEnabled) && !this.props.enabled && newProps.enabled) ) {
```

Comment:
this can be related also to the Feature component although at the moment they have different lifecycle methods. Maybe we can share the condition to do similar/same things. 
good point